src/test/scala/io/iohk/ethereum/network/rlpx/RLPxConnectionHandlerSpec.scala

```diff
@@ -70,6 +73,23 @@ class RLPxConnectionHandlerSpec extends FlatSpec with Matchers with MockFactory
     connection.expectNoMsg()
   }
 
+  it should "close the connection when Ack timeout happens" in new TestSetup {
+    (mockMessageCodec.encodeMessage _).expects(Ping(): MessageSerializable).returning(ByteString("ping encoded")).anyNumberOfTimes()
+
+    setupRLPxConnection()
+
+    rlpxConnectionParent watch rlpxConnection
+
+    rlpxConnection ! RLPxConnectionHandler.SendMessage(Ping())
+    connection.expectMsg(Tcp.Write(ByteString("ping encoded"), RLPxConnectionHandler.Ack))
+
+    //Wait for Ack timeout
+    Thread.sleep(rlpxConfiguration.waitForTcpAckTimeout.toMillis)
```

Comment:
@mcsherrylabs @ntallar actually in this case there should be a better way:

```
rlpxConnectionParent.expectTerminated(rlpxConnection, max = rlpxConfiguration.waitForTcpAckTimeout)
```

An expectation with a timeout is always better than a `Thread.sleep`.
But in general, in many places we do require those sleeps, because we lack feedback from actors under test. We would have make certain design changes (new messages) - to make those actors test-oriented.

Those sleeps are the most critical factor increasing our build times. At some point (probably not now) this issue might be worth the effort.